pkg/server/server.go

```diff
@@ -704,6 +705,11 @@ func NewServerWithOpts(opts ...OpenFGAServiceV1Option) (*Server, error) {
 		return nil, fmt.Errorf("a datastore option must be provided")
 	}
 
+	// ctx can be nil despite the default above if WithContext() was called
+	if s.ctx == nil {
+		return nil, fmt.Errorf("server cannot be started with nil context")
+	}
```

Comment:
why not put this check within `server.WithContext`?

Reply:
ah never mind, it breaks the existing pattern of putting the validations in one place